# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/atk/AccessibleWrap.cpp
# Commit: 6d0f4821ddcf
# Full Hash: 6d0f4821ddcfca1b59e4db3befc3ec9c12335891
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-03-23 09:49:32
# Regressor Bug: 1735970
# File Overlap Count: 1
# Description:
#   Bug 1735970 part 3: Add TableAccessibleBase and TableCellAccessibleBase. r=morgan
#   
#   These are needed to support RemoteAccessible tables.
#   Stuff specific to LocalAccessible is still in TableAccessible and TableCellAccessible, which now inherit from the new Base classes.
#   
# ==============================================================================

diff -r d82660fb5408 -r 6d0f4821ddcf accessible/atk/AccessibleWrap.cpp
--- a/accessible/atk/AccessibleWrap.cpp	Wed Mar 23 04:00:25 2022 +0000
+++ b/accessible/atk/AccessibleWrap.cpp	Wed Mar 23 04:00:26 2022 +0000
@@ -1548,13 +1548,13 @@
     return nullptr;
   }
 
-  AutoTArray<LocalAccessible*, 10> headerCells;
+  AutoTArray<Accessible*, 10> headerCells;
   tableCell->ColHeaderCells(&headerCells);
   if (headerCells.IsEmpty()) {
     return nullptr;
   }
 
-  return headerCells[0];
+  return headerCells[0]->AsLocal();
 }
 
 // static
@@ -1581,11 +1581,11 @@
     return nullptr;
   }
 
-  AutoTArray<LocalAccessible*, 10> headerCells;
+  AutoTArray<Accessible*, 10> headerCells;
   tableCell->RowHeaderCells(&headerCells);
   if (headerCells.IsEmpty()) {
     return nullptr;
   }
 
-  return headerCells[0];
+  return headerCells[0]->AsLocal();
 }